app/src/main/java/com/arctouch/codechallenge/home/domain/HomeMoviesUseCase.kt

```diff
@@ -0,0 +1,67 @@
+package com.arctouch.codechallenge.home.domain
+
+import com.arctouch.codechallenge.core.domain.model.*
+import com.arctouch.codechallenge.home.data.repository.GenresRepository
+import com.arctouch.codechallenge.home.data.repository.MoviesRepository
+import kotlinx.coroutines.async
+import kotlinx.coroutines.coroutineScope
+
+class HomeMoviesUseCase(
```

Comment:
Because i´m trying to follow the clean architecture principles/pattern, and thoose principles dictates that the presentation layer should never knows the data layer, all the bussiness logic must be in the usecases/interactors layer, this approach has a lot off advantages, its easier to test and do changes, it´s better to work with teams since each layer is independent also improves code reusability since i can reuse the same presentation and data layers for different apps, modules, screens, etc... just writing a new usecase for the situtation

You can read more about it here:
https://github.com/android10/Android-CleanArchitecture-Kotlin
https://github.com/ImangazalievM/CleanArchitectureManifest